rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -266,7 +266,7 @@ def _parse_step(self, step: Union[Text, Dict[Text, Any]]) -> None:
             )
         elif KEY_USER_MESSAGE in step.keys():
             self._parse_user_message(step)
-        elif KEY_USER_INTENT in step.keys() or KEY_USER_MESSAGE in step.keys():
+        elif KEY_USER_INTENT in step.keys():
             self._parse_labeled_user_utterance(step)
```

Comment:
are you sure you don't break e2e tests?

Reply:
The removed code is identical to the above `elif` statement, so totally redundant.

Reply:
currently fixing this stuff in https://github.com/RasaHQ/rasa/pull/7388/files as well